app/src/main/java/org/fossasia/pslab/activity/OscilloscopeActivity.java

```diff
@@ -271,6 +273,18 @@ public void run() {
                             }
                         }
                     }
+
+                    if (!scienceLab.isConnected() || (!isCH1Selected && !isCH2Selected && !isCH3Selected && !isMICSelected)) {
```

Comment:
Any one of them is sufficient for icon to be green. suppose device is connected and no channel is selected then also icon should be red as no data is refreshed.